### DISCUSSION TITLE
Utilizing AssemblyRunner with Azure Functions -- .NET Core Support?

### DISCUSSION THREAD

**Discussion by KristaRutz** [xunit/xunit#2245] [Question] [answered]
I was hoping to build a test runner using the AssemblyRunner class of xunit.runner.utility. My test runner relies on AssemblyRunner.WithAppDomain() as seen in the xunit.samples repo, which is supported in .NET Framework but not in .NET Core.

However, I want to run the tests from Azure Functions, which supports .NET Core 3.1 but doesn't offer any .NET Framework support. Any suggestions for how to get this functionality in the target framework - perhaps a workaround to use AssemblyRunner.WithoutAppDomain() which currently gives a `System.InvalidOperationException: Could not find/load any of the following assemblies: xunit.execution.dotnet.dll` error?

**Comment by bradwilson** [comment, marked as answer]
Make sure your unit tests are compiled against .NET Core and not .NET Framework, which should then result in the `xunit.execution.dotnet.dll` reference that's missing.
